### DISCUSSION TITLE
Override url when getting paginated results in octokit-js

### DISCUSSION THREAD

**Discussion by sparkofreason** [octokit/octokit.js#2142] [Q&A] [answered]
We have to access our instance of GHE through a proxy, which requires a port number to be added to the URL. We create an `Octokit` instance like
```
new Octokit({
    baseUrl: 'https://api.github.mycompany.com:5000',
...
}
```
We then get the page iterator as
```
Octokit.paginate.iterator(
  Octokit.rest.repos.listCommits,
  { ...options, per_page: 100 }
)
```
The first page comes through fine, but the subsequent page fails.
```
HttpError: request to https://api.github.mycompany.com/repositories/1/commits?sha=HEAD&per_page=100&page=2 failed
```
Note the lack of port. That doesn't really surprise me, since GitHub Enterprise doesn't know what port the proxy is using, of course.

Is there some way to communicate the port (or complete base URL) which will return a valid link header including the port? Or do we just have to resort to manual paging, parsing/modifying the link header as required?

**Comment by gr2m** [comment]
> We have to access our instance of GHE through a proxy

Maybe you could implement that using an http(s) Agent? You can set it using

```js
new Octokit({
  request: { agent }
})
```

See the `request.agent` option at https://github.com/octokit/request.js/#request

<img width="901" alt="image" src="https://user-images.githubusercontent.com/39992/127074847-b6aefc2a-e67f-4060-9d3e-ff1645e87d3f.png">

**Comment by sparkofreason** [comment, marked as answer]
The solution we found was to override `request.fetch` on the `Octokit` constructor options, so that the URL could be rewritten as necessary.

**Reply by gr2m** [reply]
nice hack, glad that worked for you, and thanks for sharing!
